### PR TITLE
made headings actual markdown headings

### PR DESCRIPTION
**Pull request by arne-broering**
I did this so that we can have anchor links to headings

### Description

Please provide a concise summary of the changes and their motivation.

#### Issues Addressed

List and link all the issues addressed by this PR.

#### Change Type

Please select the relevant options:

- [ ] Fix (change that resolves an issue)
- [ ] New enhancement (change that adds specification content)

#### Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document.
- [ ] My changes adhere to the established patterns, and best practices.

### CODE REVIEW

**Comment by phil-abb** [comment]
@arne-broering I believe @ajcraig intentionally changed these to not be headers (and other places) before we moved to this repo because it was causing the sidebar navigation to be too deep. Not having them as headers breaks the links so I'm not sure what is better. 

@ajcraig What do you think?

File: system-design/app-interoperability/application-package-definition.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ metadata:
     organization:
       - name: Northstar Industrial Applications
         site: http://northstar-ida.com
-deploymentProfiles:
+s:
```
Comment (phil-abb) [inline system-design/app-interoperability/application-package-definition.md]: It looks like you may have accidentally deleted some text here.

**Comment by arne-broering** [comment]
> @arne-broering I believe @ajcraig intentionally changed these to not be headers (and other places) before we moved to this repo because it was causing the sidebar navigation to be too deep. Not having them as headers breaks the links so I'm not sure what is better.
> 
> @ajcraig What do you think?

Ah, ok, didn't know that. I just changed those to actual "#" headings, as it would be really helpful to link in the type definitions from the Deployment spec to the App Package spec.

File: system-design/app-interoperability/application-package-definition.md (hunk 1 above)
Comment (arne-broering) [reply system-design/app-interoperability/application-package-definition.md]: Just fixed that

**Review by phil-abb — APPROVED** [review]

**Comment by arne-broering** [comment]
I added the following lines to `mkdocs.yml`:

```
markdown_extensions:
  - toc:
      toc_depth: 2
```

**Pull request closed without merging**